debian/rules.em

```diff
@@ -51,7 +51,7 @@ override_dh_shlibdeps:
 	# in the install tree and source it.  It will set things like
 	# CMAKE_PREFIX_PATH, PKG_CONFIG_PATH, and PYTHONPATH.
 	if [ -f "@(InstallationPrefix)/setup.sh" ]; then . "@(InstallationPrefix)/setup.sh"; fi && \
-	dh_shlibdeps -l$(CURDIR)/debian/@(Package)/@(InstallationPrefix)/lib/:$(CURDIR)/debian/@(Package)/@(InstallationPrefix)/opt/rviz_ogre_vendor/lib/
+	dh_shlibdeps -l$(CURDIR)/debian/@(Package)/@(InstallationPrefix)/lib/:$(CURDIR)/debian/@(Package)/@(InstallationPrefix)/opt/rviz_ogre_vendor/lib/:$(CURDIR)/debian/@(Package)/@(InstallationPrefix)/opt/rviz_ogre_vendor/lib/OGRE/
```

Comment:
So this change makes it so that dh_shlibdeps can find the ogre vendor. and hence finish building.  This is likely a dumb question, but I'll ask it anyway: can rviz2 find Ogre at runtime?  I just want to make sure rviz2 wasn't depending on the RPATH to find it.

Reply:
I didn't run into trouble when testing https://github.com/ros2/rviz/pull/688. I couldn't depend on those RPATHs though, those are absolute paths in the build host filesystem. In the example I posted on that PR:

``` 
$ readelf -d /opt/ros/rolling/opt/rviz_ogre_vendor/lib/OGRE/Plugin_OctreeZone.so

Dynamic section at offset 0xdd70 contains 31 entries:
  Tag        Type                         Name/Value
 0x0000000000000001 (NEEDED)             Shared library: [Plugin_PCZSceneManager.so.1.12.1]
 0x0000000000000001 (NEEDED)             Shared library: [libOgreMain.so.1.12.1]
 0x0000000000000001 (NEEDED)             Shared library: [libstdc++.so.6]
 0x0000000000000001 (NEEDED)             Shared library: [libm.so.6]
 0x0000000000000001 (NEEDED)             Shared library: [libgcc_s.so.1]
 0x0000000000000001 (NEEDED)             Shared library: [libc.so.6]
 0x000000000000000e (SONAME)             Library soname: [Plugin_OctreeZone.so.1.12.1]
 0x000000000000001d (RUNPATH)            Library runpath: [/tmp/binarydeb/ros-rolling-rviz-ogre-vendor-8.5.0/obj-x86_64-linux-gnu/ogre_install/lib:/tmp/binarydeb/ros-rolling-rviz-ogre-vendor-8.5.0/obj-x86_64-linux-gnu/ogre_install/lib/OGRE]
...
```
you can see `/tmp/binarydeb/ros-rolling-rviz-ogre-vendor-8.5.0/obj-x86_64-linux-gnu/` paths.

Reply:
> can rviz2 find Ogre at runtime? I just want to make sure rviz2 wasn't depending on the RPATH to find it.

The environment hook supplied by the vendor package provides the library location for all other use cases. https://github.com/ros2/rviz/tree/ros2/rviz_ogre_vendor/env_hook

Reply:
It looks like these plugins are vendored within the vendored ogre package. Since the environment hook doesn't add this subdirectory presumably these plugins are only ever linked by ogre itself and it has configuration to know where to find them.

Otherwise how would e.g. rviz know to find libraries in `/opt/ros/rolling/lib/rviz_ogre_vendor/lib/OGRE` vs /opt/ros/rolling/lib/rviz_ogre_vendor/lib`?

Reply:
> The environment hook supplied by the vendor package provides the library location for all other use cases. https://github.com/ros2/rviz/tree/ros2/rviz_ogre_vendor/env_hook

Ah, cool.  Thanks for that link, that makes sense.

One other dumb question: why is the env_hook setting the `LD_LIBRARY_PATH` to only `PREFIX/opt/rviz_ogre_vendor/lib`, while this change is additionally adding `PREFIX/opt/rviz_ogre_vendor/lib/OGRE`?

Reply:
That hook doesn't point to the plugin libraries. I haven't found what pulls those in.

Reply:
All right, that's fair.  Thanks for the feedback.  I have no objection to the patch, so I'm going to approve.  It would be good to get another approval from @nuclearsandwich before merging.